js/src/data/actions.js

```diff
@@ -52,7 +52,7 @@ export function* fetchShippingRates() {
 			return {
 				countryCode: el.country_code,
 				currency: el.currency,
-				rate: el.rate.toString(),
+				rate: Number( el.rate ),
```

Comment:
The shipping rates and times returned from API are numeric values in string type. And move the type casting from `reducer` to `actions` for separating the responsibility that `reducer` focus on how to maintain the state tree; `actions` focus on how to fetch data and transform it to client-side suitable structures for further use.

Reply:
Agree thanks!

Reply:
❓ 
Are we sure we don't need a float? Because in the previous reducer I can see `parseFloat` who is telling us that we want a float (maybe wrongly).

Reply:
> Are we sure we don't need a float? Because in the previous reducer I can see `parseFloat` who is telling us that we want a float (maybe wrongly).

We still need float values for shipping rates. The floating-point value is also the primitive type `number` in JavaScript. Use Number() to casting a floating-point numeric `string` to a `number` is supported, and it usually has more properly fault-tolerant and reasonable.

Reply:
Understood!